### PR TITLE
fix(ui): Avoid setting a default value if none is set to boolean inputs

### LINKED ISSUE #1904: [UI] Boolean input always have a value
### Expected Behavior

Boolean inputs should be able to have no value.

### Actual Behaviour

When executing a flow that have a boolean input from the UI, the execution will always have a value for the input even if it has a default value or is not mandatory.
This is because the UI uses a Switch with true/false so no-value or undefined didn't exist.

As the input is always send, if it's optional or with a default value of true, it will be set to false which is not what we want.

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version:
- Operating System (OS / Docker / Kubernetes):
- Java Version (If not docker):

### Example flow

Launching this flow from the UI will output false in the logs whereas true should be outputted as the default of the input is true.

```
id: hello-world
namespace: dev
inputs:
  - name: clear_landing_before
    type: BOOLEAN
    required: true
    defaults: true
tasks:
  - id: echo_input
    type: io.kestra.core.tasks.debugs.Echo
    format: "{{inputs.clear_landing_before}}"
```

### PR DESCRIPTION
**Pull request by Skraye**
Now this will keep undefined if no default is set and if user does not touch the switch to set any value for the boolean inputs.

close #1904

### CODE REVIEW

File: ui/src/components/flows/FlowRun.vue [hunk 1]
```diff
@@ -149,9 +150,7 @@
             for (const input of this.flow.inputs || []) {
                 this.inputs[input.name] = input.defaults;
 
-                if(input.type === "BOOLEAN" && input.defaults){
-                    this.inputs[input.name] = (/true/i).test(input.defaults);
-                }
+                console.log("this.inputs[input.name] end:", this.inputs[input.name])
```
Comment (loicmathieu) [inline ui/src/components/flows/FlowRun.vue:153]: Avoid console logging (twice)

**Review by loicmathieu — COMMENTED** [review]
How would it work if the user wants to set the boolean to false? As the switch is by default in false so the user will not touch it so it would lead to undefined?

**Comment by Skraye** [comment]
> How would it work if the user wants to set the boolean to false? As the switch is by default in false so the user will not touch it so it would lead to undefined?

If a default value is set, then it will be set up automatically,

No default > undefined
Default true > switch is set to true
Default false > switch is set to false

**Comment by loicmathieu** [comment]
So when there is no default and the user want to set it to false what will happen?

**Comment by Skraye** [comment]
Then he have to switch twice
As its a switch I cannot have three state, so its either you have one by default, or its undefined by default
At best I can add a warning or something like this

Or have you a suggestion to face that ?

**Comment by loicmathieu** [comment]
> Or have you a suggestion to face that ?

No, this is really the issue here, we need 3-state for boolean as one can have a default or an optional value

**Comment by tchiotludo** [comment]
already done here: https://github.com/kestra-io/kestra/pull/2184

**Comment by tchiotludo** [comment]
ok don't read properly, the issue is still relevant and I agree with @loicmathieu true / false is not the same than null, we should handle it, the best is to use this [component](https://element-plus.org/en-US/component/radio.html#button-style)

**Comment by Skraye** [comment]
Done :arrow_down: 
image

**Comment by brian-mulier-p** [comment]
defaults: false still shows as undefined
image
image

File: ui/src/components/flows/FlowRun.vue [hunk 2]
```diff
@@ -146,8 +150,8 @@
             for (const input of this.flow.inputs || []) {
                 this.inputs[input.name] = input.defaults;
 
-                if (input.type === "BOOLEAN" && input.defaults){
-                    this.inputs[input.name] = (/true/i).test(input.defaults) || input.defaults === true;
+                if (input.type === "BOOLEAN" && !input.defaults){
```
Comment (brian-mulier-p) [inline ui/src/components/flows/FlowRun.vue:153]:
```suggestion
                if (input.type === "BOOLEAN" && input.defaults === undefined){
```

File: ui/src/components/flows/FlowRun.vue [hunk 3]
```diff
@@ -36,10 +36,14 @@
...
                 />
-                <el-switch
+                <el-radio-group
                     v-if="input.type === 'BOOLEAN'"
                     v-model="inputs[input.name]"
-                />
+                >
+                    <el-radio-button label="true" />
+                    <el-radio-button label="false" />
+                    <el-radio-button label="undefined"/>
```
Comment (brian-mulier-p) [inline ui/src/components/flows/FlowRun.vue:45]:
```suggestion
                    <el-radio-button :label="$t('true')" />
                    <el-radio-button :label="$t('false')" />
                    <el-radio-button :label="$t('undefined')"/>
```

**Review by brian-mulier-p — CHANGES_REQUESTED** [review]

**Pull request merged**